cmd/snap-confine/snap-confine.c

```diff
@@ -738,20 +731,10 @@ static void enter_non_classic_execution_environment(sc_invocation * inv,
 	// This simplifies testing if any processes belonging to a given snap are
 	// still alive as well as to properly account for each application and
 	// service.
-	if (getegid() != 0 && saved_gid == 0) {
-		// Temporarily raise egid so we can chown the freezer cgroup under LXD.
-		if (setegid(0) != 0) {
-			die("cannot set effective group id to root");
-		}
-	}
+	// Temporarily raise egid so we can chown the freezer cgroup under LXD.
```

Comment:
I think this should be moved into the `if() { ... }` below and rephrased to something like:
```
// egid may be temporarily raised inside when creating relevant cgroups
```

Reply:
I removed the comment, the place that now handles this operation already had a comment I added in the patches above. Thanks for spotting.